packages/contracts/contracts/minter-suite/Minters/MinterSEAV1.sol

```diff
@@ -0,0 +1,1169 @@
+ * artist configures their project on this minter, or when a new auction is
+ * started. The purpose of this token is to allow users to have a preview of
+ * the next token that will be auctioned, even before the auction has started.
+ * @notice Privileged Roles and Ownership:
+ * This contract is designed to be managed, with limited powers.
+ * Privileged roles and abilities are controlled by the core contract's Admin
+ * ACL contract and a project's artist. Both of these roles hold extensive
+ * power and can modify minter details.
+ * Care must be taken to ensure that the admin ACL contract and artist
+ * addresses are secure behind a multi-sig or other access control mechanism.
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to the core contract's Admin ACL
+ * contract:
+ * - ejectNextTokenTo
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to this minter's minter filter's
+ * Admin ACL contract:
+ * - updateMinterTimeBufferSeconds
+ * - updateRefundGasLimit
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to a project's artist:
+ * - syncProjectMaxInvocationsToCore
+ * - manuallyLimitProjectMaxInvocations
+ * - configureFutureAuctions
+ * - tryPopulateNextToken
+ * ----------------------------------------------------------------------------
+ * The following functions are restricted to a project's artist or the core
+ * contract's Admin ACL contract:
+ * - resetFutureAuctionDetails
+ * ----------------------------------------------------------------------------
+ * Additional admin and artist privileged roles may be described on other
+ * contracts that this minter integrates with.
+ *
+ * @dev Note that while this minter makes use of `block.timestamp` and it is
+ * technically possible that this value is manipulated by block producers, such
+ * manipulation will not have material impact on the ability for collectors to
+ * place a bid before auction end time. This is due to the admin-configured
+ * `minterTimeBufferSeconds` parameter, which will used to ensure that
+ * collectors have sufficient time to place a bid after the final bid and
+ * before the auction end time.
+ */
+contract MinterSEAV1 is ReentrancyGuard, ISharedMinterV0, ISharedMinterSEAV0 {
+    using SafeCast for uint256;
+
+    /// Minter filter address this minter interacts with
+    address public immutable minterFilterAddress;
+
+    /// Minter filter this minter may interact with.
+    IMinterFilterV1 private immutable minterFilter;
+
+    /// minterType for this minter
+    string public constant minterType = "MinterSEAV1";
+
+    /// minter version for this minter
+    string public constant minterVersion = "v1.0.0";
+
+    uint256 constant ONE_MILLION = 1_000_000;
+
+    uint256 public constant MIN_AUCTION_DURATION_SECONDS = 60; // seconds
```

Comment:
note: hard-coding here results in a subgraph e2e test taking >60s. Alternatives could be to remove guardrail entirely, or make it defined in constructor.

Reply:
Hard-coding this value seems totally fine based on intended usage of this minter, and ease of upgradeability in the future in our new minter suite architecture that we're moving toward.